### PR TITLE
[CoW] Migrate Batches Table

### PR DESCRIPTION
**Pull request by bh2smith**
MIgrates Cow protocol batches spell for both mainnet and gnosis chain.

**Note that** - this is based on #3745 !

cc @gentrexha

### CODE REVIEW

**Review by gentrexha — APPROVED** [review]
Left some discussion comments. LGTM.

Thanks for migrating all these queries Ben!

File: models/cow_protocol/ethereum/cow_protocol_ethereum_batches.sql [hunk 1]
```diff
@@ -69,29 +70,19 @@ combined_batch_info as (
...
-               THEN (
-                select count(*)
-                from {{ ref('dex_trades') }}
-                where tx_hash = evt_tx_hash
-                and blockchain = 'ethereum'
-              )
-            ELSE dex_swaps
-        END                                              as dex_swaps,
+        (
+            select count(*) from dex.trades
```
[the remark this answers is not in the document]
Comment (bh2smith) [reply models/cow_protocol/ethereum/cow_protocol_ethereum_batches.sql:74]: Thats exactly what we want - this was one of the main reasons we introduced the separate tables. We only want dex swaps here.

File: models/cow_protocol/ethereum/cow_protocol_ethereum_batches.sql [hunk 2]
```diff
@@ -69,29 +70,19 @@ combined_batch_info as (
...
+        )                                                as dex_swaps,
         batch_value,
         solver                                           as solver_address,
         evt_tx_hash                                      as tx_hash,
         gas_price,
         gas_used,
         (gas_price * gas_used * eth_price) / pow(10, 18) as tx_cost_usd,
         fee_value,
-        length(data)::decimal / 1024                     as call_data_size,
+        2 * bytearray_length(data) / 1024                     as call_data_size,
```
[the remark this answers is not in the document]
Comment (bh2smith) [reply models/cow_protocol/ethereum/cow_protocol_ethereum_batches.sql:85]: bytearray_length is not the same as the number of characters like we had before.
So, excluding `0x`

```
length(ffff) = 4 = 2 * bytearray_length(ffff)
```

File: models/cow_protocol/gnosis/cow_protocol_gnosis_batches.sql [hunk 3]
```diff
@@ -67,25 +68,16 @@ combined_batch_info as (
...
-               OR name ilike '%ParaSwap'
-               OR name ilike '%0x'
-               OR name = 'Legacy'
-              )
-              -- Estimation made here: https://dune.com/queries/1646084
-              THEN ((gas_used - 73688 - (70528 * num_trades)) / 90000)::int
-            ELSE dex_swaps
-        END                                              as dex_swaps,
+        -- Estimation made here: https://dune.com/queries/1646084
+        (gas_used - 73688 - (70528 * num_trades)) / 90000 as dex_swaps,
```
[the remark this answers is not in the document]
Comment (bh2smith) [reply models/cow_protocol/gnosis/cow_protocol_gnosis_batches.sql:72]: In this query you use to disprove the approximation, you are assuming that the dex.trades table accounts for all dex trades (which is not necessarily true). If we are going to assume that as a source of truth, then why not just use that are our value here instead...

File: models/cow_protocol/gnosis/cow_protocol_gnosis_batches.sql [hunk 4]
```diff
@@ -67,25 +68,16 @@ combined_batch_info as (
...
         batch_value,
         solver                                           as solver_address,
         evt_tx_hash                                      as tx_hash,
         gas_price,
         gas_used,
         (gas_price * gas_used * eth_price) / pow(10, 18) as tx_cost_usd,
         fee_value,
-        length(data)::decimal / 1024                     as call_data_size,
+        2 * bytearray_length(data) / 1024                     as call_data_size,
```
[the remark this answers is not in the document]
Comment (bh2smith) [reply models/cow_protocol/gnosis/cow_protocol_gnosis_batches.sql:80]: See above answer.

here is example:

https://dune.com/queries/2731355


Furthermore if you take a hex string and save it to a file and check the size kb you will see they match

File: models/cow_protocol/gnosis/cow_protocol_gnosis_batches.sql (hunk 3 above)
[the remark this answers is not in the document]
Comment (gentrexha) [reply models/cow_protocol/gnosis/cow_protocol_gnosis_batches.sql:72]: That's a good point.

I think it's still worthwhile to be able to validate this approach somehow and this was a straight approach for me.

[the remark this answers is not in the document]
Comment (bh2smith) [reply models/cow_protocol/gnosis/cow_protocol_gnosis_batches.sql:72]: I do think that it's more likely dex trade is missing data so I can't really take these comparisons as true. What matters most to us is knowing is dex swaps is zero or not zero and I would be interested to see just how often we see disagreement just for that relaxed condition.

File: models/cow_protocol/gnosis/cow_protocol_gnosis_batches.sql (hunk 4 above)
[the remark this answers is not in the document]
Comment (bh2smith) [reply models/cow_protocol/gnosis/cow_protocol_gnosis_batches.sql:80]: See above answer.

here is example:

https://dune.com/queries/2731355


Furthermore if you take a hex string and save it to a file and check the size kb

**Pull request merged**